scripts/hal_alignment/hal_gene_liftover.py

```diff
@@ -0,0 +1,156 @@
+    """
+    seq_region_regex = re.compile(
+        '^(?P<chrom>[^:]+):(?P<start>[0-9]+)-(?P<end>[0-9]+):(?P<strand>1|-1)$'
+    )
+    match = seq_region_regex.match(region)
+
+    try:
+        region_chrom = match['chrom']  # type: ignore
+        match_start = match['start']  # type: ignore
+        match_end = match['end']  # type: ignore
+        match_strand = match['strand']  # type: ignore
+    except TypeError as e:
+        raise ValueError(f"region '{region}' could not be parsed") from e
+
+    region_start = int(match_start) - 1
+    region_end = int(match_end)
+    region_strand = '-' if match_strand == '-1' else '+'
+
+    if region_start >= region_end:
+        raise ValueError(f"region '{region}' has inverted/empty interval")
+
+    return SimpleRegion(region_chrom, region_start, region_end, region_strand)
+
+
+if __name__ == '__main__':
+
+    parser = ArgumentParser()
+    parser.add_argument('hal_file', help="Input HAL file.")
+    parser.add_argument('src_genome', help="Source genome name.")
+    parser.add_argument('dest_genome', help="Destination genome name.")
+    parser.add_argument('output_file', help="Output file.")
+
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument('--src-region', help="Region to liftover.")
+    group.add_argument('--src-bed-file', help="BED file containing regions to liftover.")
+
+    parser.add_argument('--flank', default=0, type=int,
+                        help="Requested length of upstream/downstream"
+                             " flanking regions to include in query.")
+
+    args = parser.parse_args()
+
+    hal_file = args.hal_file
+    src_genome = args.src_genome
+    dest_genome = args.dest_genome
+    output_file = args.output_file
+    src_region = args.src_region
+    src_bed_file = args.src_bed_file
+    flank = args.flank
+
+
+    if flank < 0:
+        raise ValueError(f'invalid flank length: {flank}')
+
+    with TemporaryDirectory() as tmp_dir_name:
+        tmp_dir = Path(tmp_dir_name)
+
+        query_bed_file = tmp_dir / 'src_regions.bed'
+
+        if src_region is not None:
+            src_regions = [parse_region(src_region)]
+        else:  # i.e. bed_file is not None
+            src_regions = pybedtools.BedTool(src_bed_file)
+
+        make_src_region_file(src_regions, query_bed_file, flank_length=flank)
+
+        cmd1 = ['halLiftover', '--outPSL', hal_file, src_genome, query_bed_file, dest_genome,
+                'stdout']
+        cmd2 = ['pslPosTarget', 'stdin', output_file]
+        with Popen(cmd1, stdout=PIPE) as p1:
+            with Popen(cmd2, stdin=p1.stdout) as p2:
+                p2.wait()
+                if p2.returncode != 0:
+                    raise RuntimeError(f'pslPosTarget terminated with signal {-p2.returncode}')
+            p1.wait()
+            if p1.returncode != 0:
+                raise RuntimeError(f'halLiftover terminated with signal {-p1.returncode}')
```

Comment:
I'm not sure why do you need such a complex setup with Pipes? It feels like you could do the same via an intermediate file in a simpler manner (you already have the temporary folder). If it works it can be left as it is, just want to understand the design reasons behind it.

Reply:
I wanted to avoid the creation of the intermediate file, though perhaps this is being overly parsimonious with storage space. But I can see how code readability might suffer for it, since the 8 lines used to run these 2 processes could be replaced with 2 consecutive calls to `subprocess.run`. If you think the code would be clearer as a result, I would be happy to use an intermediate file instead.

Reply:
It is fine by me as it is, but I will leave it to @dthybert to decide what he would prefer. Unless the files are expected to surpass the 2GB threshold (in total), you should be fine :)

Reply:
For the gene set of a single genome it's unlikely that the temporary files would need more than 2GB, though I haven't rigorously tested this.

Reply:
Yes this was the part I was unsure in term of code structure the PR.  I understand why you did this in term of storage place it is better and I guess in performance as well as there is no need for parsing the file on disk.  However it is more because I did not came across this before.  As this sound technically better than having an intermediary file I would keep this like that. Maybe you could add a comment to show what it does in shell : "cmd1 | cmd2"